core/src/main/scala/com/yahoo/maha/core/query/QueryPipeline.scala

```diff
@@ -951,49 +980,14 @@ OuterGroupBy operation has to be applied only in the following cases
       val factBestCandidateOption =
         requestModel.bestCandidates.map(_ => findBestFactCandidate(requestModel, forceDisqualifyEngine, dimEngines))
       val dimensionCandidates = findBestDimCandidates(requestModel, factBestCandidateOption.map(_.fact.engine).getOrElse(OracleEngine), dimensionCandidatesMapping).filter(db => {
-        val queryGenerator = queryGeneratorRegistry.getGenerator(db.dim.engine)
+        val queryGenerator = queryGeneratorRegistry.getGenerator(db.dim.engine, None)
```

Comment:
This doens't look right.  Need to verify the engine constraints on generator and you want to do it on the version of generator used.

Reply:
I realized that, but at this point, engine is not finalized so we cannot select query generator buckets, and hence don't know the version. 

We can select buckets (and versions) for all engines at the beginning but then bucket percentages will not mean what they should, because all requests will result in random number generation for all engines, whether that engine is used or not.

Another option is to move validateEngineConstraints() to the Engine objects (HiveEngine, OracleEngine etc), because it seems to be engine specific, not query generator specific.